#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/apartment_structure/protos/Wall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/WoodenBox.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5557626464605794 0.5720881060741148 0.6031940646986386 2.090535281931896
  position 0.7889297927251969 -1.354297578162365 27.038224517313985
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 14 14
  floorTileSize 2 2
  wallHeight 0.3
}
WoodenBox {
  translation 1.96 -5.61 0.3
}
Wall {
  rotation -1 0 0 -5.307179586466759e-06
  name "wall(2)"
  size 0.1 4 1
}
Wall {
  translation 1.5 2 0
  rotation -0.8944271909999159 0.4472135954999579 0 -5.307179586466759e-06
  name "wall(1)"
  size 3 0.1 1
}
Wall {
  translation 1.5 -2 0
  rotation -1 0 0 -5.307179586466759e-06
  name "wall(6)"
  size 3 0.1 1
}
Robot {
  translation 2 0 0.0084604
  rotation 0.013354002234353124 0.008562721432689845 0.9998741672960609 -3.0537953071795862
  children [
    DEF Body Transform {
      translation 0 0 0.11
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
          }
          geometry Cylinder {
            height 0.2
            radius 0.15
          }
        }
      ]
      translationStep 0
      rotationStep 0
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -3117.2045323575717
        anchor 0.15 0 0.05
      }
      device [
        RotationalMotor {
          name "motor1"
        }
      ]
      endPoint Solid {
        translation 0.14103535773023002 0.00028949826568602907 0.049850300464676235
        rotation 0.5688127091481104 0.6133208521158678 0.5479868924266827 2.050385889270344
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.1 0.1 0.1
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 8854.552422463155
        anchor -0.15 0 0.05
      }
      device [
        RotationalMotor {
          name "motor2"
        }
      ]
      endPoint Solid {
        translation -0.14108997401484236 -4.9539049434897054e-05 0.04985572324596037
        rotation 0.7027654545845371 0.10188596634940812 0.7040880383188144 2.9531912833334193
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF Front Solid {
      translation 0 -0.13 0.21
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
          }
          geometry Box {
            size 0.02 0.04 0.01
          }
        }
      ]
      name "front"
    }
    Solid {
      translation 0.1 0.1 0.21
      rotation 0 1 0 0
      children [
        DistanceSensor {
          rotation 0 0 1 0.7854003061004253
          name "distance sensor_E_2"
          lookupTable [
            0 0 0
            3 3 0
          ]
          aperture 1
          gaussianWidth 1.57
        }
        Shape {
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "Sensor_E_2"
    }
    Solid {
      translation -0.1 0.1 0.21
      children [
        DistanceSensor {
          rotation 0 0 1 2.3562
          name "distance sensor_D_2"
          lookupTable [
            0 0 0
            3 3 0
          ]
          aperture 1
          gaussianWidth 1.5697
        }
        Shape {
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "Sensor_D_2"
    }
    Solid {
      translation -0.1 -0.1 0.21
      children [
        DistanceSensor {
          rotation 0 0 1 -2.355
          name "distance sensor_D_1"
          lookupTable [
            0 0 0
            3 3 0
          ]
          aperture 1
          gaussianWidth 1.57
        }
        Shape {
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "Sensor_D_1"
    }
    Solid {
      translation 0.1 -0.1 0.21
      children [
        DistanceSensor {
          rotation 0 0 1 -0.7854
          name "distance sensor_E_1"
          lookupTable [
            0 0 0
            3 3 0
          ]
          aperture 1
          gaussianWidth 1.57
        }
        Shape {
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "Sensor_E_1"
    }
    Solid {
      translation 1.26827e-08 4.42916e-09 0.21
      rotation 0 0 1 3.14
      children [
        Compass {
          name "compass_sensor"
        }
        GPS {
          name "gps_sensor"
        }
        Shape {
          appearance PBRAppearance {
            baseColor 0.333333 0.341176 0.32549
            roughness 1
          }
          geometry Cylinder {
            height 0.02
            radius 0.04
          }
        }
      ]
      name "GPS and Compass"
      boundingObject Cylinder {
        height 0.02
        radius 0.04
      }
      recognitionColors [
        0.333333 0.341176 0.32549
      ]
    }
  ]
  name "MB-01"
  boundingObject USE Body
  physics Physics {
  }
  controller "my_controller_Micael"
}
